### PR TITLE
FileSystem class added filter method close #8

### LINKED ISSUE #8: Move the `filter_by_extension` function to the `FileSystem`
https://github.com/LikaloLLC/docsie-universal-doc-importer/blob/35a95e77d7dcac7ff5ce18e76ac935b96d07fe78/universal_doc_importer/utils.py#L1

Move the whole function to the `FileSystem` class as it's related to it and can not be used anywhere.

### PR DESCRIPTION
**Pull request by Zarif99**
Close #8

### CODE REVIEW

File: universal_doc_importer/file_system.py [hunk 1]
```diff
@@ -1,34 +1,45 @@
 class FileSystem():
-    def __init__(self, filePath=None):
+    def __init__(self, extensions, filePath=None):
```
Comment (NikBelyaev) [inline universal_doc_importer/file_system.py]:
```
def __init__(self, filePath=None, extensions=None):
    self.extensions = extensions or ['md']
```

File: universal_doc_importer/file_system.py [hunk 2]
```diff
@@ -37,10 +48,64 @@ def printAllChildren(self, depth=-1):
                 child.printAllChildren(depth)
             print("\t" * depth + "}")
 
-    def makeDict(self):
-        dictionary = {self.name: []}
+    def buildDict(self):
```
Comment (NikBelyaev) [inline universal_doc_importer/file_system.py]: rename it to `_buildDict`, as it is the core functionality of the `buidDict`.

File: universal_doc_importer/file_system.py [hunk 3]
```diff
@@ -37,10 +48,64 @@ def printAllChildren(self, depth=-1):
...
         if len(self.children) > 0:
+            dictionary = {self.name: []}
             for child in self.children:
-                dictionary[self.name].append(child.makeDict())
+                dictionary[self.name].append(child.buildDict())
+            return dictionary
+        else:
+            return self.name
+
+    def makeDict(self):
```
Comment (NikBelyaev) [inline universal_doc_importer/file_system.py]: Rename it to `buildDict`

File: universal_doc_importer/file_system.py [hunk 4]
```diff
@@ -37,10 +48,64 @@ def printAllChildren(self, depth=-1):
...
+        :rtype: dict:
+        """
         if len(self.children) > 0:
+            dictionary = {self.name: []}
             for child in self.children:
-                dictionary[self.name].append(child.makeDict())
+                dictionary[self.name].append(child.buildDict())
+            return dictionary
+        else:
+            return self.name
```
Comment (1mace1) [inline universal_doc_importer/file_system.py:64]: return dict anyway, `{self.name: []}`

File: universal_doc_importer/file_system.py [hunk 5]
```diff
@@ -37,10 +48,64 @@ def printAllChildren(self, depth=-1):
...
+        :return: data:
+        :rtype: dict:
+        """
+        #  Get data from buildDict method and check if data is empty
+        #  it returns data like this { repo_name : []}
+        data = self.buildDict()
+        repo_name = '/'.join(list(data)[0].split('-')[:2])
+        result = self.filter_by_extension(data)
+        if result == {}:
+            result[repo_name] = []
```
Comment (1mace1) [inline universal_doc_importer/file_system.py:79]: Once the `_buildDict` is returning dict anyway, you don't need this. self.name should be filled with the repo name somewhere in the `__init__`

**Review by NikBelyaev — CHANGES_REQUESTED** [review]

**Comment by NikBelyaev** [comment]
Also, resolve merge conflicts

**Review by NikBelyaev — APPROVED** [review]

**Pull request merged**